library/Requests.php

```diff
@@ -259,13 +259,14 @@ public static function trace($url, $headers = array(), $options = array()) {
 	 * @param array $headers
 	 * @param array $data
 	 * @param array $options
+	 * @param array $files
 	 * @return Requests_Response
 	 */
 	/**
 	 * Send a POST request
 	 */
-	public static function post($url, $headers = array(), $data = array(), $options = array()) {
-		return self::request($url, $headers, $data, self::POST, $options);
+	public static function post($url, $headers = array(), $data = array(), $options = array(), $files = array()) {
```

Comment:
IMO, this should either be part of `$data` or part of `$options`. For the former, I wouldn't be opposed to having a `Requests_File` interface (with a default implementation) so that you could do e.g:

```
$response = Requests::post(
    'http://example.com',
    [ 'Content-Type' => 'text/plain' ],
    new Requests_File( __DIR__ . '/README.txt' )
);
```

(And potentially allow an array of these.)

Reply:
That's a good idea, would allow us to set filenames and mime-types, just like cURL_File does, I was originally intending to do it as part of data, then remembered how this library is based on the Python Requests library and thought that following their convention would be appreciated.

If not, I have nothing against wrapping into a `Requests_File` instead.

Reply:
Python has the benefit of named parameters, so in Python it's:
```py
requests.post( 'http://example.com', files={ 'file': open( 'x', 'rb' ) } )
```

The equivalent for PHP is our `$options` array, but we already deviate a bit anyway due to the nature of the language, so I think overloading the `$data` parameter makes more sense.

Reply:
True, makes sense. I think `Requests_File` is the more elegant way of doing it then. I'll work on this a bit and send more stuff into here. Thanks.

Reply:
btw, for the interface, I'd recommend calling it `Requests_File_Interface`, and having `get_headers()` and `get_body()` methods, which should allow an advanced implementation to offer multipart uploads (although maybe not in Requests itself) or uploading using a data string.

The default implementation should be `Requests_File`, and we should be able to imply most stuff (upload name, size, etc) from the filename passed in.